client-keycloak/src/main/java/org/apache/atlas/keycloak/client/AbstractKeycloakClient.java

```diff
@@ -0,0 +1,149 @@
+package org.apache.atlas.keycloak.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.apache.atlas.AtlasErrorCode;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.keycloak.client.config.KeycloakConfig;
+import org.apache.atlas.keycloak.client.service.AtlasKeycloakAuthService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+import org.springframework.lang.NonNull;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
+import static org.apache.atlas.AtlasErrorCode.BAD_REQUEST;
+import static org.apache.atlas.AtlasErrorCode.RESOURCE_NOT_FOUND;
+
+abstract class AbstractKeycloakClient {
+
+    private final static Logger LOG = LoggerFactory.getLogger(AbstractKeycloakClient.class);
+    private static final Map<Integer, AtlasErrorCode> ERROR_CODE_MAP = new HashMap<>();
+
+    private static final int DEFAULT_KEYCLOAK_RETRY = 3;
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String BEARER = "Bearer ";
+
+    protected final KeycloakConfig keycloakConfig;
+    protected final RetrofitKeycloakClient retrofit;
+
+    private final AtlasKeycloakAuthService authService;
+    private String currentToken;
+
+    static {
+        ERROR_CODE_MAP.put(HTTP_NOT_FOUND, RESOURCE_NOT_FOUND);
+        ERROR_CODE_MAP.put(HTTP_BAD_REQUEST, BAD_REQUEST);
+    }
+
+    public AbstractKeycloakClient(KeycloakConfig keycloakConfig) {
+        this.keycloakConfig = keycloakConfig;
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
```

Comment:
Remove the interceptor if not used, we can also remove logging-interceptor dependency if not used